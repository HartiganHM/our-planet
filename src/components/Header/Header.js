import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import * as actions from '../../actions';
import './Header.css';

class Header extends Component {
  componentDidMount() {
    this.props.populateData();
  }

  render() {
    return (
      <div className="Header">
        <NavLink to="/">
          <img src="../images/Our-Planet-Text.png" alt="our-planet-wordmark" />
        </NavLink>
        <NavLink to="/about">About</NavLink>
      </div>
    );
  }
}

export const mapStateToProps =() => {

}

export const mapDispatchToProps = (dispatch) => {
  return {
    populateData: () => dispatch(actions.fetchAllData)
  }
}

export default connect(null, mapDispatchToProps)(Header);
