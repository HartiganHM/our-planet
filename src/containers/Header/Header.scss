//Layout, Box, Visual, Typography, Misc
@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';
.Header {
    @include padding(1.5);
    border-bottom: $simple-border;
    background-color: $snow;
    .wrapper {
        display: flex;
        justify-content: space-between;
        align-items: middle;
        @include width(160);
        margin: auto;
        .headerLogo {
            flex: 1;
            @include height(4.5);
            @include width(30);
            background: {
                image: url('../../images/branding/Our-Planet-Wordmark-Small.png');
                repeat: no-repeat;
                size: contain;
            }
        }
        .nav-bar {
            display: flex;
            align-items: center;
            .nav-button {
                // @include margin(0.5);
                @include padding(0.5);
                @include width(10);
                border: $simple-border;
                color: $accent-color-dark;
                font-size: 1.2rem;
                text: {
                    align: center;
                    decoration: none;
                }
                transition: 0.1s background-color,
                color ease-in-out;
                &:hover {
                    background-color: $accent-color-light;
                    color: $snow;
                }
            }
            .active {
                border-color: $accent-color-dark;
                background-color: $accent-color-dark;
                color: $snow;
                cursor: default;
                &:hover {
                    background-color: $accent-color-dark;
                }
            }
        }
    }
}

@media screen and (max-width: 1634px) {
    .Header {
        .wrapper {
            @include width(128);
        }
    }
}

@media screen and (max-width: 1334px) {
    .Header {
        .wrapper {
            @include width(96);
        }
    }
}

@media screen and (max-width: 1014px) {
    .Header {
        .wrapper {
            flex-direction: column;
            align-items: center;
            @include width(64);
            .headerLogo {
                margin-bottom: 15px;
            }
        }
    }
}

@media screen and (max-width: 694px) {
    .Header {
        .wrapper {
            @include width(32);
        }
    }
}