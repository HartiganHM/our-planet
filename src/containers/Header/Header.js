import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';
import * as actions from '../../actions';
import PropTypes from 'prop-types';
import './Header.css';

export class Header extends Component {
  componentDidMount() {
    this.props.populateData();
  }

  render() {
    const { location } = this.props;
    const aboutActive =
      location === '/about' ? 'nav-button active' : 'nav-button';
    const homeActive = location === '/' ? 'nav-button active' : 'nav-button';

    return (
      <div className="Header">
        <div className="wrapper">
          <span className="headerLogo" />
          <div className="nav-bar">
            <NavLink exact to="/" className={homeActive}>
              Home
            </NavLink>
            <NavLink exact to="/about" className={aboutActive}>
              About
            </NavLink>
          </div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = store => {
  return {
    animals: store.animals
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    populateData: () => dispatch(actions.fetchAllData()),
    resetAnimals: (inputValue, animals) =>
      dispatch(actions.searchAnimals(inputValue, animals))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);

Header.propTypes = {
  animals: PropTypes.array,
  populateData: PropTypes.func,
  resetAnimals: PropTypes.func,
  location: PropTypes.string
};
