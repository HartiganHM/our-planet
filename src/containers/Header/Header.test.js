import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapDispatchToProps } from './Header';

describe('Header tests', () => {
  it('Should match the snapshot', () => {
    const mockPopulateData = jest.fn();
    const renderedHeader = shallow(<Header populateData={mockPopulateData} />);
    expect(renderedHeader).toMatchSnapshot();
  });

  describe('Header mapDispatchToProps tests', () => {
    it('Should call dispatch when populateData is called', () => {
      const mockDispatch = jest.fn();
      const result = mapDispatchToProps(mockDispatch);

      result.populateData();
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
