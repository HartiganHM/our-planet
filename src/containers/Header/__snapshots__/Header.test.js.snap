// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    populateData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wrapper"
>
        <span
                className="headerLogo"
        />
        <div
                className="nav-bar"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-button"
                        exact={true}
                        to="/our-planet/"
                >
                        Home
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-button"
                        exact={true}
                        to="/our-planet/about"
                >
                        About
                </NavLink>
        </div>
</div>,
      "className": "Header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="headerLogo"
/>,
          <div
            className="nav-bar"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-button"
                        exact={true}
                        to="/our-planet/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-button"
                        exact={true}
                        to="/our-planet/about"
            >
                        About
            </NavLink>
</div>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "headerLogo",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-button"
                exact={true}
                to="/our-planet/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-button"
                exact={true}
                to="/our-planet/about"
>
                About
</NavLink>,
            ],
            "className": "nav-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav-button",
                "exact": true,
                "to": "/our-planet/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "About",
                "className": "nav-button",
                "exact": true,
                "to": "/our-planet/about",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrapper"
>
          <span
                    className="headerLogo"
          />
          <div
                    className="nav-bar"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav-button"
                              exact={true}
                              to="/our-planet/"
                    >
                              Home
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav-button"
                              exact={true}
                              to="/our-planet/about"
                    >
                              About
                    </NavLink>
          </div>
</div>,
        "className": "Header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="headerLogo"
/>,
            <div
              className="nav-bar"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="nav-button"
                            exact={true}
                            to="/our-planet/"
              >
                            Home
              </NavLink>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="nav-button"
                            exact={true}
                            to="/our-planet/about"
              >
                            About
              </NavLink>
</div>,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "headerLogo",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-button"
                  exact={true}
                  to="/our-planet/"
>
                  Home
</NavLink>,
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="nav-button"
                  exact={true}
                  to="/our-planet/about"
>
                  About
</NavLink>,
              ],
              "className": "nav-bar",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Home",
                  "className": "nav-button",
                  "exact": true,
                  "to": "/our-planet/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "About",
                  "className": "nav-button",
                  "exact": true,
                  "to": "/our-planet/about",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
